### DISCUSSION TITLE
Improve documentation for move_group_interface vs. MoveitCpp

### DISCUSSION THREAD

**Discussion by kappa95** [moveit/moveit2#2255] [Ideas]
I have some difficulties in understanding what I can achieve with MoveItCpp with respect to move_group_interface C++. Which are the differences, which are the pros and cons for both. If the MoveitCpp can't still substitute move_group_interface can I use both together?

Also, which are the config files that I have to change for using one API with respect to other, which are the best practice?

**Comment by bmaxdk** [comment]
For `MoveItCpp` allows more precise control. Since it does not rely on action client, the performance is slightly better. However, for complex to work if you go more deeper.

For `move_group_interface`, it is stable interface. This interface provides a simple to high-level API to interface with the move group. which also making it easier to use for common task. However, this interface might not offer the same level of control and customization as `MoveItCpp`.

If you are using with standard application, go with `move_group_interface`. If you want to implement advanced control and customization a new capability within **MoveIt**, then use with `MoveItCpp`.

Hope it helps
